daemon/daemon/docker_watcher.go

```diff
@@ -200,136 +201,92 @@ func (d *Daemon) updateProbeLabels(dockerID string) (bool, *types.Container, err
 		return false, nil, fmt.Errorf("Error while inspecting container '%s': %s", dockerID, err)
 	}
 
-	ciliumLabels := labels.Labels{}
+	newLabels := labels.Labels{}
 	if dockerCont.Config != nil {
 		log.Debugf("Read docker labels %+v", dockerCont.Config.Labels)
-		ciliumLabels = d.getFilteredLabels(dockerCont.Config.Labels)
-	}
-	log.Debugf("Using filtered labels %+v", ciliumLabels)
-
-	return d.updateOperationalLabels(dockerID, dockerCont, ciliumLabels, true)
-}
-
-func (d *Daemon) updateUserLabels(dockerID string, labels labels.Labels) (bool, *types.Container, error) {
-	dockerCont, err := d.dockerClient.ContainerInspect(ctx.Background(), dockerID)
-	if err != nil {
-		return false, nil, fmt.Errorf("Error while inspecting container '%s': %s", dockerID, err)
+		newLabels = d.getFilteredLabels(dockerCont.Config.Labels)
 	}
-	return d.updateOperationalLabels(dockerID, dockerCont, labels, false)
-}
+	log.Debugf("Using filtered labels %+v", newLabels)
 
-func (d *Daemon) updateOperationalLabels(dockerID string, dockerCont dTypes.ContainerJSON, newLabels labels.Labels, isProbe bool) (bool, *types.Container, error) {
-	isNewContainer := false
-	var (
-		cont           types.Container
-		epLabelsSHA256 string
-	)
-
-	if ciliumContainer, ok := d.containers[dockerID]; !ok {
-		isNewContainer = true
-		cont = types.Container{
+	c, ok := d.containers[dockerID]
+	if !ok {
+		cont := types.Container{
 			ContainerJSON: dockerCont,
+			NRetries:      0,
 			OpLabels: labels.OpLabels{
-				AllLabels:      newLabels.DeepCopy(),
-				UserLabels:     labels.Labels{},
-				ProbeLabels:    newLabels.DeepCopy(),
-				EndpointLabels: newLabels.DeepCopy(),
+				Custom:        labels.Labels{},
+				Disabled:      labels.Labels{},
+				Orchestration: newLabels.DeepCopy(),
 			},
-			NRetries: 0,
 		}
+		d.containers[dockerID] = &cont
+
+		return true, &cont, nil
+	}
+
+	if c.NRetries > maxRetries {
+		d.DeleteIdentityBySHA256(c.OpLabels.Enabled().SHA256Sum(), c.ID)
+		return false, nil, nil
+	}
+
+	if !d.EndpointExists(endpoint.DockerUUIDPrefix + ":" + c.ID) {
```

Comment:
I'm not exactly sure how to do this long term hence the open coding. I think this should be done as we converge onto a single hash map with a single lookup function. Not sure yet.